src/stdlib/dns_lookup.rs

```diff
@@ -12,10 +15,69 @@ mod non_wasm {
     use domain::resolv::stub::conf::{ResolvConf, ResolvOptions, ServerConf, Transport};
     use domain::resolv::stub::Answer;
     use domain::resolv::StubResolver;
+    use once_cell::sync::Lazy;
+    use tokio::runtime::Handle;
 
     use crate::compiler::prelude::*;
     use crate::value::Value;
 
+    static WORKER: Lazy<Worker> = Lazy::new(Worker::new);
+
+    type Job<T> = Box<dyn FnOnce() -> T + Send + 'static>;
+    struct JobHandle<T> {
+        job: Job<T>,
+        result: Arc<mpsc::Sender<T>>,
+    }
+
+    struct Worker {
+        thread: Option<thread::JoinHandle<()>>,
+        queue: Option<mpsc::Sender<JobHandle<Result<Answer, Error>>>>,
+    }
+
+    impl Worker {
+        fn new() -> Self {
+            let (sender, receiver) = mpsc::channel::<JobHandle<Result<Answer, Error>>>();
+            let receiver = Arc::new(Mutex::new(receiver));
+            Self {
+                thread: Some(thread::spawn(move || loop {
+                    match receiver.lock().unwrap().recv() {
+                        Ok(handle) => {
+                            let result = (handle.job)();
+                            handle.result.as_ref().send(result).unwrap();
+                        }
+                        Err(_) => todo!(),
+                    }
+                })),
+                queue: Some(sender),
+            }
+        }
+
+        fn execute<F>(&self, f: F) -> Result<Answer, Error>
+        where
+            F: FnOnce() -> Result<Answer, Error> + Send + 'static,
+        {
+            let job = Box::new(f);
+            let (sender, receiver) = mpsc::channel();
+            let receiver = Arc::new(Mutex::new(receiver));
+            let handle = JobHandle {
+                job,
+                result: Arc::new(sender),
+            };
```

Comment:
Similarly to my above question, did you borrow this from somewhere? It seems like a lot of "work" to do for each execute function such that I'm wondering what the performance looks like. Maybe we could add a benchmark for it in `benches/stdlib.rs` to see?

Reply:
There are already some benchmarks there for the `dns_lookup` function. Did you mean we should add some benchmarks for the `Worker` itself?